### PR TITLE
Version 7.0.0

### LINKED ISSUE #515: Enhance documentation: service configuration, client init
The documentation should be expanded with regard service initialization and general init configuration of the client.
Details regarding `monaco-vscode-api` services should either be described in more detail here or links to a more detailed description should be added.

- [x] Document services exposed via `monaco-vscode-api`: https://github.com/CodinGame/monaco-vscode-api#monaco-standalone-services

Improve README:
  - [x] Make clear/document at an obvious place that enabling themes (and textmate) service prevents using monarch grammars (will come with #564)
  - [x] Explain minimal usage of `initService` and explain what it means (will come with #564)

### PR DESCRIPTION
**Pull request by kaisalmen**
- Revised the `treemending` approach. The postinstall step is removed. `monaco-languageclient` no longer patches and existing `monaco-editor` instead the package `@codingame/monaco-editor-treemended` is used. This requires that projects using this lib have to enforce the correct editor by overrides (npm/pnpm) or resolutions (yarn) in the `package.json`.
- Updated to v9 of `vscode-languageclient` and `vscode-languageserver`
- `vscode-ws-jsonrpc` will be updated to `3.1.0` because the underlying `vscode-jsonrpc` is updated to `8.2.0`

TODO:
- [x] Local tests
- [x] Release new next releases
  - https://www.npmjs.com/package/monaco-languageclient/v/7.0.0-next.0
  - https://www.npmjs.com/package/monaco-languageclient-examples/v/7.0.0-next.0
  - https://www.npmjs.com/package/vscode-ws-jsonrpc/v/3.1.0-next.0
- Update README
  - [x] Highlight breaking changes / make instructions more prominent
  - [x] Restructure it, e.g. move project history to the end

Closes #515

### CODE REVIEW

**Comment by kaisalmen** [comment]
@CGNonofr @dkattan @BusinessDuck please test the new next versions if you have the time (this should resolve #550) . The only thing missing is the documentation update

**Comment by dkattan** [comment]
> @CGNonofr @dkattan @BusinessDuck please test the new next versions if you have the time (this should resolve #550) . The only thing missing in this PR is the documentation update, then it will ready for review.

Sure thing. My issue currently is I am upgrading from v4 and I now have significant regressions in Monaco functionality. For example bracket matching no longer works and it seems to randomly stop having syntax highlighting. I made a few other changes recently so I'm trying to isolate variables but I'm mentioning it in case you feel it's related or know a quick fix.

**Comment by kaisalmen** [comment]
> I made a few other changes recently so I'm trying to isolate variables but I'm mentioning it in case you feel it's related or know a quick fix.

@dkattan one thing, you could try to re-produce some of the problems with our examples in this branch. For example, if they are language server implementation independent and are just related to monao-editor/service configuration.

**Comment by BusinessDuck** [comment]
oh, i missed up this, on my way! let you know about results

**Comment by BusinessDuck** [comment]
```
ERROR in ./node_modules/.pnpm/@codingame+monaco-vscode-api@1.83.5/node_modules/@codingame/monaco-vscode-api/vscode/src/vs/workbench/services/workspaces/common/workspaceTrust.js 322:19-39
export 'isTemporaryWorkspace' (imported as 'isTemporaryWorkspace') was not found in 'monaco-editor/esm/vs/platform/workspace/common/workspace.js' (possible exports: EXTENSION_DEVELOPMENT_EMPTY_WINDOW_WORKSPACE, IWorkspaceContextService, STANDALONE_EDITOR_WORKSPACE_ID, UNKNOWN_EMPTY_WINDOW_WORKSPACE, WORKSPACE_EXTENSION, WORKSPACE_FILTER, Workspace, WorkspaceFolder, isEmptyWorkspaceIdentifier, isSingleFolderWorkspaceIdentifier, isStandaloneEditorWorkspace, isWorkspaceIdentifier, toWorkspaceIdentifier)

ERROR in ./node_modules/.pnpm/@codingame+monaco-vscode-api@1.83.5/node_modules/@codingame/monaco-vscode-api/vscode/src/vs/workbench/services/workspaces/common/workspaceTrust.js 486:16-48
export 'IWorkspaceTrustEnablementService' (imported as 'IWorkspaceTrustEnablementService') was not found in 'monaco-editor/esm/vs/platform/workspace/common/workspaceTrust.js' (possible exports: IWorkspaceTrustManagementService)
```

**Comment by BusinessDuck** [comment]
do we still need aliasing? "vscode": "npm:@codingame/monaco-vscode-api@1.83.5",

**Comment by kaisalmen** [comment]
> do we still need aliasing? "vscode": "npm:@codingame/monaco-vscode-api@1.83.5",

@BusinessDuck yes or we could also use `overrides` (npm/pnpm) or `resoutions` (yarn).
Did you set any `overrides` instructions in your package.json?
What's the output of `pnpm list`?

**Comment by kaisalmen** [comment]
I will overhaul the README here today

**Comment by BusinessDuck** [comment]
> @BusinessDuck yes or we could also use `overrides` (npm/pnpm) or `resoutions` (yarn). 
can you clarify please, we still need vscode overrides ?
> Did you set any `overrides` instructions in your package.json? What's the output of `pnpm list`?
Only for vscode api

**Comment by kaisalmen** [comment]
For `monaco-editor` you require this to override any mismatching version in the dependency hierachy:
```yaml
  "overrides": {
    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.83.5 <1.84.0"
  },
  "resolutions": {
    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.83.5 <1.84.0"
  }
```
I bet you have mismatching versions. What is the output of `pnpm list`?

**Comment by BusinessDuck** [comment]
I dont have directly dependency `monaco-editor`, i trying to use `@codingame/monaco-editor-treemended` instead of base monaco-editor package. Is that wrong? Should i revert to base package and add override and resolution ?

**Comment by kaisalmen** [comment]
> Is that wrong? Should i revert to base package and add override and resolution ?

No, it is not, but you need to override any other references with the overrides. If `pnpm list --depth 2 monaco-editor` reveals nothing you are good. You should only see output here: `pnpm list --depth 2 @codingame/monaco-editor-treemended`

**Comment by BusinessDuck** [comment]
Offtopic question: As i can see you're guys (you're awesome), have added `localizeWithPath` function. That mean i can import `setLocale` function. But typings isn't works corrects fro that import and i need hacky d.ts:

```
declare module 'monaco-editor/esm/vs/nls' {
	export function setLocale(local: unknown, data: unknown): void;
}
```

**Comment by kaisalmen** [comment]
> But typings isn't works corrects fro that import and i need hacky d.ts:

@CGNonofr any idea ⬆️

**Comment by BusinessDuck** [comment]
That typings problem actually big one, take a look at this my custom d.ts

```typescript
// NOTE this modules is just for make this typings work

declare module 'monaco-editor/esm/vs/nls' {
	export function setLocale(local: unknown, data: unknown): void;
}

declare module 'monaco-editor/esm/vs/base/common/keybindings' {
	// eslint-disable-next-line @typescript-eslint/consistent-type-imports
	import type { KeyCode } from 'monaco-editor';

	export interface Modifiers {
		readonly ctrlKey: boolean;
		readonly shiftKey: boolean;
		readonly altKey: boolean;
		readonly metaKey: boolean;
	}

	export function decodeKeybinding(keybinding: number | number[], os: number): Keybinding | null;
	export interface Chord extends Modifiers {
		keyCode: KeyCode;
		getHashCode(): string;
		isModifierKey(): boolean;
		toKeybinding(): Keybinding;
		isDuplicateModifierCase(): boolean;
	}
	export interface Keybinding {
		chords: Chord[];
	}
}

declare module 'monaco-editor/esm/vs/base/common/keyCodes' {
	// eslint-disable-next-line @typescript-eslint/consistent-type-imports
	import type { KeyCode } from 'monaco-editor';

	export namespace KeyCodeUtils {
		export function toString(keyCode: KeyCode): string;
	}
}

declare module 'monaco-editor/esm/vs/platform/keybinding/common/keybindingsRegistry' {
	import { Keybinding } from 'monaco-editor/esm/vs/base/common/keybindings';

	type ContextKeyExpression = unknown;
	export interface IKeybindingItem {
		keybinding: Keybinding | null;
		command: string | null;
		// tslint:disable-next-line:no-any
		commandArgs?: any;
		when: ContextKeyExpression | null | undefined;
		weight1: number;
		weight2: number;
		extensionId: string | null;
		isBuiltinExtension: boolean;
	}
	export interface IKeybindingsRegistry {
		getDefaultKeybindings(): IKeybindingItem[];
	}
}

declare module 'monaco-editor/esm/vs/platform/registry/common/platform' {
	export interface IRegistry {

		/**
		 * Adds the extension functions and properties defined by data to the
		 * platform. The provided id must be unique.
		 * @param id a unique identifier
		 * @param data a contribution
		 */
		// tslint:disable-next-line:no-any
		add(id: string, data: any): void;

		/**
		 * Returns true iff there is an extension with the provided id.
		 * @param id an extension identifier
		 */
		knows(id: string): boolean;

		/**
		 * Returns the extension functions and properties defined by the specified key or null.
		 * @param id an extension identifier
		 */
		as<T>(id: string): T;
	}

	export const Registry: IRegistry;
}

declare module 'monaco-editor/esm/vs/platform/actions/common/actions' {
	export interface EditorCommonActions {
		MenubarEditMenu:
			| 'undo'
			| 'redo'
			| 'editor.action.clipboardCutAction'
			| 'editor.action.clipboardCopyAction'
			| 'editor.action.clipboardPasteAction'
			| 'editor.action.commentLine'
			| 'editor.action.blockComment'
			| 'actions.find'
			| 'editor.action.startFindReplaceAction';
		CommandPalette:
			| 'undo'
			| 'redo'
			| 'editor.action.selectAll'
			| 'editor.action.clipboardCutAction'
			| 'editor.action.clipboardCopyAction'
			| 'editor.action.clipboardPasteAction';
		MenubarSelectionMenu:
			| 'editor.action.selectAll'
			| 'editor.action.copyLinesUpAction'
			| 'editor.action.copyLinesDownAction'
			| 'editor.action.duplicateSelection'
			| 'editor.action.moveLinesUpAction'
			| 'editor.action.moveLinesDownAction'
			| 'editor.action.insertCursorAbove'
			| 'editor.action.insertCursorBelow'
			| 'editor.action.insertCursorAtEndOfEachLineSelected'
			| 'editor.action.addSelectionToNextFindMatch'
			| 'editor.action.addSelectionToPreviousFindMatch'
			| 'editor.action.selectHighlights'
			| 'editor.action.smartSelect.expand'
			| 'editor.action.smartSelect.shrink';
		MenubarGoMenu:
			| 'editor.action.jumpToBracket'
			| 'editor.action.revealDefinition'
			| 'editor.action.revealDeclaration'
			| 'editor.action.goToTypeDefinition'
			| 'editor.action.goToImplementation'
			| 'editor.action.goToReferences'
			| 'editor.action.marker.nextInFiles'
			| 'editor.action.marker.prevInFiles';
		EditorContext:
			| 'editor.action.clipboardCutAction'
			| 'editor.action.clipboardCopyAction'
			| 'editor.action.clipboardPasteAction'
			| 'editor.action.refactor'
			| 'editor.action.sourceAction'
			| 'editor.action.revealDefinition'
			| 'editor.action.revealDeclaration'
			| 'editor.action.goToTypeDefinition'
			| 'editor.action.goToImplementation'
			| 'editor.action.goToReferences'
			| 'editor.action.formatDocument'
			| 'editor.action.formatSelection'
			| 'editor.action.changeAll'
			| 'editor.action.rename'
			| 'editor.action.quickOutline'
			| 'editor.action.quickCommand';
		SimpleEditorContext:
			| 'editor.action.clipboardCutAction'
			| 'editor.action.clipboardCopyAction'
			| 'editor.action.clipboardPasteAction';
		EditorContextPeek:
			| 'editor.action.peekDefinition'
			| 'editor.action.peekDeclaration'
			| 'editor.action.peekTypeDefinition'
			| 'editor.action.peekImplementation'
			| 'editor.action.referenceSearch.trigger';
		gotoErrorTitleMenu: 'editor.action.marker.next' | 'editor.action.marker.prev';
		suggestWidgetStatusBar:
			| 'acceptSelectedSuggestion'
			| 'acceptSelectedSuggestion'
			| 'acceptSelectedSuggestion'
			| 'acceptAlternativeSelectedSuggestion'
			| 'acceptAlternativeSelectedSuggestion'
			| 'toggleSuggestionDetails'
			| 'toggleSuggestionDetails';
		Navigation: 'workbench.action.showCommands';
	}

	export type EditorCommonActionsKey = keyof EditorCommonActions;

	export namespace MenuRegistry {
		interface MenuId {
			id: EditorCommonActionsKey;
		}

		interface Node {
			element: {
				command: {
					id: string;
					title: string;
				};
			};
			next: Node;
		}

		interface LinkedList {
			_first: Node;
			_last: Node;
			_size: number;
			_remove: (node: Node) => void;
		}
		export const _menuItems: Map<MenuId, LinkedList>;
	}
}
```

I need that because we using imports like this:

```
import { decodeKeybinding } from 'monaco-editor/esm/vs/base/common/keybindings';
import { setLocale } from 'monaco-editor/esm/vs/nls';
import { MenuRegistry } from 'monaco-editor/esm/vs/platform/actions/common/actions';
```

**Comment by kaisalmen** [comment]
@CGNonofr This is ready for review now. Please check the README. I moved some content around and re-wrote "Using monaco-languageclient"

New next version is also available:
https://www.npmjs.com/package/monaco-languageclient/v/7.0.0-next.1

**Comment by kaisalmen** [comment]
@dkattan and @BusinessDuck feel free to check this PR, too. Especially if the README makes things clear.

**Comment by BusinessDuck** [comment]
Everything is working well via pnpm, after all updates, that you describe. Thanks! I need some time to test UI, as i can see the `command palette` is down :( 

```
standaloneServices.js:325 Uncaught (in promise) Error: command 'editor.action.quickCommand' not found
```

**Comment by BusinessDuck** [comment]
```
commandsQuickAccess.js:67 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'preserveInput')
    at get configuration [as configuration] (commandsQuickAccess.js:67:1)
    at CommandsQuickAccessProvider.updateOptions (commandsQuickAccess.js:76:1)
    at new CommandsQuickAccessProvider (commandsQuickAccess.js:62:1)
    at InstantiationService._createInstance (instantiationService.js:97:1)
    at InstantiationService.createInstance (instantiationService.js:68:1)
    at QuickAccessController.getOrInstantiateProvider (quickAccess.js:186:1)
    at QuickAccessController.doShowOrPick (quickAccess.js:41:1)
    at QuickAccessController.show (quickAccess.js:36:1)
    at ShowAllCommandsAction.run (commandsQuickAccess.js:228:1)
    at handler (actions.js:384:1)
    ```
    
    When i change the event name to `workbench..action.showCommands`

**Comment by dkattan** [comment]
> @dkattan and @BusinessDuck feel free to check this PR, too. Especially if the README makes things clear.

So far so good with Yarn. I am getting the following warnings, but I have been getting them since adding the resolutions into my package:

```
warning Pattern ["vscode@npm:@codingame/monaco-vscode-api@>=1.83.5 <1.84.0"] is trying to unpack in the same destination "C:\\Users\\DKattan\\AppData\\Local\\Yarn\\Cache\\v6\\npm-vscode-1.83.5-1af675b15ddf643cf394c66f853dd498c21c92c3-integrity\\node_modules\\vscode" as pattern ["vscode@npm:@codingame/monaco-vscode-api@1.83.5"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["vscode@npm:@codingame/monaco-vscode-api"] is trying to unpack in the same destination "C:\\Users\\DKattan\\AppData\\Local\\Yarn\\Cache\\v6\\npm-vscode-1.83.5-1af675b15ddf643cf394c66f853dd498c21c92c3-integrity\\node_modules\\vscode" as pattern ["vscode@npm:@codingame/monaco-vscode-api@1.83.5"]. This could result in non-deterministic behavior, skipping.
warning chart.js@4.4.0: The engine "pnpm" appears to be invalid.
warning vscode-languageclient@9.0.1: The engine "vscode" appears to be invalid.
warning vscode-languageclient@7.0.0: The engine "vscode" appears to be invalid.
[3/4] Linking dependencies...
warning " > monaco-languageclient@7.0.0-next.1" has incorrect peer dependency "monaco-editor@0.44.0".
```

This appears to be benign.

File: README.md [hunk 1]
```diff
@@ -42,87 +35,42 @@ Click [here](https://www.typefox.io/blog/teaching-the-language-server-protocol-t
...
+Earlier in 2023 we started to treemend an existing `monaco-editor` dependency via a postinstall script. This adds back monaco-editor code that was removed during bundling/threeshaking (*treemending*). See the detailed explanation [here](https://github.com/CodinGame/monaco-vscode-api#why). But, this introduced multiple problems.
 
-### Monaco-editor / @codingame/monaco-vscode-api compatibility table
+#### Overrides instructions
 
-The following table describes which version of **monaco-languageclient** and **@codingame/monaco-vscode-api** are compatible with a specific version of **monaco-editor**. The listing starts with version 2.0.0 because **@codingame/monaco-vscode-api** was introduced for the first time.
+With v7 we decided to use readily treemended version if monaco-editor called [@codingame/monaco-editor-treemended](https://www.npmjs.com/package/@codingame/monaco-editor-treemended), but this requires to add `overrides` (npm/pnmpm) and `resolutions` (yarn) in your project. Setting these ensures that all dependencies to `monaco-editor` and `vscode` are aligned:
```
Comment (dkattan) [inline README.md:60]: if monaco-editor -> of monaco-editor

File: README.md [hunk 2]
```diff
@@ -42,87 +35,42 @@ Click [here](https://www.typefox.io/blog/teaching-the-language-server-protocol-t
...
 
-**Important:** Due to the [treemending](#treemending) mentioned above, it is mandatory you use the correct monaco-editor version. This is defined by peerDependency in [@codingame/monaco-vscode-api](https://github.com/CodinGame/monaco-vscode-api)
+```yaml
```
Comment (dkattan) [inline README.md:62]: Shouldn't this be json instead of yaml?

File: README.md [hunk 3]
```diff
@@ -236,17 +184,17 @@ npm run start:verify:webpack
...
 
 ## Troubleshooting
 
 ### General
 
-If you use **monaco-languageclient** make sure you have a version of **monaco-editor** installed in your project that is compliant with **monaco-languageclient** and its peer dependency [@codingame/monaco-vscode-api](https://github.com/CodinGame/monaco-vscode-api).
+Again, if you use **monaco-languageclient** make sure you define matching [Overrides instructions](#overrides-instructions) in your local project to override any mismatching **monaco-editor** or **vscode** versions with you dependency tree.
 
-Ensure **monaco-editor** and **monaco-languageclient** are imported before you do any **monaco-editor** intialization. This ensures `monaco` and `vscode` (from **@codingame/monaco-vscode-api**) are imported beforehand. This is for example done like this in all examples contained in this repository.
+Ensure **monaco-editor**, **vscode** and **monaco-languageclient** are imported before you do any **monaco-editor** or vscode-api intialization or start using it.
```
Comment (dkattan) [inline README.md:197]: I'm dumb at JavaScript, can we give an example of the wrong and right way to import these items?

**Comment by BusinessDuck** [comment]
in previous version for restore command palette configuration should be

```
// install Monaco language client services, need to run monaco module after monaco-treemending
await monacoLsp.initServices({
  configureEditorOrViewsServiceConfig: {},
  enableQuickaccessService: true,
  enableModelService: true,
  enableKeybindingsService: true,
  enableLanguagesService: true,
});
```

how it works now, can you clarify in documentation?

I check typing but its not described. And command pallete do not works now for me everything else is ok.

```
  export interface IEditorOverrideServices {
        [index: string]: any;
    }    
```

**Comment by dkattan** [comment]
> > @dkattan and @BusinessDuck feel free to check this PR, too. Especially if the README makes things clear.
> 
> So far so good with Yarn. I am getting the following warnings, but I have been getting them since adding the resolutions into my package:
> 
> ```
> warning Pattern ["vscode@npm:@codingame/monaco-vscode-api@>=1.83.5 <1.84.0"] is trying to unpack in the same destination "C:\\Users\\DKattan\\AppData\\Local\\Yarn\\Cache\\v6\\npm-vscode-1.83.5-1af675b15ddf643cf394c66f853dd498c21c92c3-integrity\\node_modules\\vscode" as pattern ["vscode@npm:@codingame/monaco-vscode-api@1.83.5"]. This could result in non-deterministic behavior, skipping.
> warning Pattern ["vscode@npm:@codingame/monaco-vscode-api"] is trying to unpack in the same destination "C:\\Users\\DKattan\\AppData\\Local\\Yarn\\Cache\\v6\\npm-vscode-1.83.5-1af675b15ddf643cf394c66f853dd498c21c92c3-integrity\\node_modules\\vscode" as pattern ["vscode@npm:@codingame/monaco-vscode-api@1.83.5"]. This could result in non-deterministic behavior, skipping.
> warning chart.js@4.4.0: The engine "pnpm" appears to be invalid.
> warning vscode-languageclient@9.0.1: The engine "vscode" appears to be invalid.
> warning vscode-languageclient@7.0.0: The engine "vscode" appears to be invalid.
> [3/4] Linking dependencies...
> warning " > monaco-languageclient@7.0.0-next.1" has incorrect peer dependency "monaco-editor@0.44.0".
> ```
> 
> This appears to be benign.

I read the updated README.md and realized I was missing the "vscode" resolution and I had a direct dependency to `"vscode": "npm:@codingame/monaco-vscode-api",`

 I updated my resolutions to match what you have in the README.md and it no longer shows the unpack warnings.

This is the output when running `yarn --force`

```
[1/4] Resolving packages...
[2/4] Fetching packages...
warning chart.js@4.4.0: The engine "pnpm" appears to be invalid.
warning vscode-languageclient@9.0.1: The engine "vscode" appears to be invalid.
warning vscode-languageclient@7.0.0: The engine "vscode" appears to be invalid.
[3/4] Linking dependencies...
warning " > monaco-languageclient@7.0.0-next.1" has incorrect peer dependency "monaco-editor@0.44.0".
[4/4] Rebuilding all packages...
success Saved lockfile.
Done in 13.23s.
```

**Comment by BusinessDuck** [comment]
My working setup.

```
"devDependencies": {
    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.83.4 <1.84.0",
    "vscode": "npm:@codingame/monaco-vscode-api@>=1.83.4 <1.84.0",
},

 "overrides": {
      "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.83.5 <1.84.0"
    },
    "resolutions": {
      "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.83.4 <1.84.0"
    }

```

File: README.md (hunk 2 above)
Comment (CGNonofr) [reply README.md:62]: (json is a valid yaml though)

Comment (kaisalmen) [reply README.md:62]: using "```json" sometimes gives strange red lines. "```yaml" just works. 🙂

**Comment by CGNonofr** [comment]
> > But typings isn't works corrects fro that import and i need hacky d.ts:
> 
> @CGNonofr any idea ⬆️

monaco-editor only exports a few types of its public api and not every module types, you're not really supposed to import those file by hands. That's why in monaco-vscode-api we use vscode types to import monaco-editor modules

regarding `monaco-editor/esm/vs/nls`, it was not designed to be used by hands but via the `monaco-vscode-language-package-*` packages instead but it can be exported if there is the need for it

File: README.md (hunk 3 above)
Comment (kaisalmen) [reply README.md:197]: Yes, adding an example code snippet here or referring to an existing example in the repo is a good idea.

File: README.md (hunk 1 above)
Comment (kaisalmen) [reply README.md:60]: Thanks, I will change that.

File: README.md (hunk 3 above)
Comment (kaisalmen) [reply README.md:197]: I will refer/link to our python language client example.

**Comment by kaisalmen** [comment]
@CGNonofr and @dkattan I have updated the README. Thanks for the feedback, so far. I have included two images that demonstrate the overrides effect on the dependency tree.

> how it works now, can you clarify in documentation?

@BusinessDuck The command palette works with doing anything now because it it is pre-configured:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/client/src/monaco-vscode-api-services.ts#L62-L74

But, yes, you are right. This should be documented as well. Will add something to the README.

**Comment by kaisalmen** [comment]
@CGNonofr  ⬆️ Further README restructuring and now there is a sub-chapter explaining service usage. I need to add one point from  #515 regarding textmate / monarch XOR and then the README is in a nice shape, IMO.

**Comment by kaisalmen** [comment]
Done. ✅ ⬆️ Everything is ready now. @CGNonofr do you see the need for any further changes in the Troubleshooting section of the README

**Review by CGNonofr — APPROVED** [review]
LGTM, let's add additional troubleshooting section later when required

**Comment by kaisalmen** [comment]
@CGNonofr we iterate from here. 🙂 

Are we ready for v7 release then? I guess there will be new issues because we enforce our own monaco-editor now, but I think it is well documented now. I will add links from the CHANGELOG to the README.

**Comment by CGNonofr** [comment]
> @CGNonofr we iterate from here. 🙂
> 
> Are we ready for v7 release then? I guess there will be new issues because we enforce our own monaco-editor now, but I think it is well documented now. I will add links from the CHANGELOG to the README.

I guess so! it will help a lot pnpm users for sure. I'm ready for brand new error reports (:

**Pull request merged**